File: algorithm_practice_python/leetcode/arrays101/1528.py
from typing import List


class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        ans = [''] * len(s)

        for k, v in zip(indices, s):
            ans[k] = v

        return "".join(ans)


if __name__ == '__main__':
    s = "codeleet"
    indices = [4, 5, 6, 7, 0, 2, 1, 3]
    ans = Solution().restoreString(s, indices)
    print(f'{ans}, leetcode')

File: algorithm_practice_typescript/1528.ts
function restoreString(s: string, indices: number[]): string {
    const ans: string[] = []
    for (let i = 0; i <= indices.length; i++) {
        ans[indices[i]] = s[i]
    }
    return ans.join("")
};

const s = "codeleet"
const indices = [4,5,6,7,0,2,1,3]

console.log(restoreString(s, indices))